### PR TITLE
Bluetooth: Controller: Fix jitter in offsets for Extended and Periodic Advertising

### PR DESCRIPTION
**Pull request by cvinayak**
Updated ticker implementation to return remainder value for
a ticker when enumerating active tickers with time
reservations.

This is required to find offsets and to use the remainder
value to correctly calculate auxiliary offsets to the
microsecond resolution.

Use macro to adjust ticks for jitter stored in remainder
value.

Use remainder value in scheduling the periodic auxiliary
PDUs and use the ticker next slot get interface with
remainder value to fill the auxiliary offsets with
microsecond precision in the primary channel PDUs.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h [hunk 1]
```diff
@@ -32,16 +38,30 @@
 		/ 1000UL \
 	)
 
+/* Macro to remove ticks and return positive remainder value in microseconds */
+#define HAL_TICKER_REMOVE_JITTER(t, r) \
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h:42]: Why macro? Wouldn't a function be better option? It does a type checking.

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h [hunk 2]
```diff
@@ -32,16 +38,30 @@
...
+	{ \
+		if ((!(r / 1000000UL)) || (r & BIT(31))) { \
+			t--; \
+			r += 30517578UL; \
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h:46]: Please change a magic number into named constant. The same value is repeated in multiple places hence it is a great candidate to be named constant.

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h [hunk 3]
```diff
@@ -22,7 +28,7 @@
 		& HAL_TICKER_CNTR_MASK \
 	)
 
-/* Macro returning remainder in nanoseconds */
+/* Macro returning remainder in picoseconds */
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h:31]: Was it an error in former description of the macro?

File: subsys/bluetooth/controller/ll_sw/lll_adv.h [hunk 4]
```diff
@@ -129,10 +127,11 @@ struct lll_adv_aux {
 	 */
 	uint16_t data_chan_counter;
 
-	/* Temporary stored use by primary channel PDU event to fill the
+	/* Store used by primary channel PDU event to fill the
 	 * auxiliary offset to this auxiliary PDU event.
 	 */
 	uint32_t ticks_offset;
+	uint32_t us_offset;
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/lll_adv.h:134]: These members names does not say anything about reference point for these offsets. Hence one is forced to get to struct and luckily read the comment above. I suggest to change these names.
```suggestion
	uint32_t prim_adv_ticks_offset;
	uint32_t prim_adv_us_offset;
```

**Review by ppryga-nordic — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/Kconfig.ll_sw_split [hunk 5]
```diff
@@ -198,7 +198,7 @@ config BT_CTLR_ADV_PDU_LINK
 
 config BT_CTLR_ADV_PDU_BACK2BACK
 	bool "Back-to-back transmission of extended advertising trains"
-	depends on BT_CTLR_ADV_EXT && BT_BROADCASTER
+	depends on BT_BROADCASTER && BT_CTLR_ADV_EXT
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/Kconfig.ll_sw_split:204]: just being consistent with the hierarchy of features

**Review by kruithofa — APPROVED** [review]

**Review by ppryga-nordic — APPROVED** [review]

**Pull request merged**